system/core/database.php

```diff
@@ -599,10 +599,40 @@ function __construct($filename) {
 		function lock() {
 			if ($this->fd)
 				throw new LogicException('Cannot lock file ' . $this->filename . ', already locked');
-			$this->fd = fopen($this->filename, 'c+');
-			if ($this->fd === false)
-				throw new RuntimeException('Cannot lock file ' . $this->filename . ', open failed: ' . error_get_last()['message']);
-			flock($this->fd, LOCK_EX);
+
+			// Clear the last error, to prevent using an
+			// older error when something goes wrong without
+			// generating a PHP error (e.g. like read/write
+			// before PHP 7.4).
+			error_clear_last();
+
+			for ($attempt = 0; $attempt < 10; ++$attempt) {
```

Comment:
Magic number, overweeg deze als `MAX_NUMBER_ATTEMPTS` oid te implementeren

Reply:
Mjah, het is een waarde die heel lokaal is, dus er een class constant van maken is wellicht wat overdone, maar het is wel een magic number... Ik heb er nu maar een lokale "constante" van gemaakt, dan heeft het iig een naam. Hoe lijkt dat?

Reply:
Perfect